### DISCUSSION TITLE
DDS Quality

### DISCUSSION THREAD

**Discussion by GraPhiX-Guru** [dlemstra/Magick.NET#1051] [Help]
Hello
I was wondering if someone could help me with the DDS export from Magick.net there seems to be 'banding' that i cannot get rid of? for example my source image is a png
![front](https://user-images.githubusercontent.com/35724756/136415229-3615c4f5-5950-46f6-86d9-1af6c52efed8.png)

if i convert to DDS using BC7 compression in paint.net i get this, which is correct
![PNDDS](https://user-images.githubusercontent.com/35724756/136415699-d8b8296a-5eb3-4c0e-91a2-8fa3529ab1bb.png)

if i convert with my app using magick.net same format & compression DDS BC7 i get this, notice the banding?
![IMDDS](https://user-images.githubusercontent.com/35724756/136415901-666c28cb-c666-43b9-a008-51e04ad3fcb4.png)

How can i export without the banding ?

here is my code snippet:

Using FrontImage As New MagickImage(FRONT_FILE)
            Export_Format = MagickFormat.Dds
            FrontImage.Settings.Compression = CompressionMethod.BC7
            FrontImage.Settings.SetDefine("dds:mipmaps", "0")
            FrontImage.Settings.SetDefine("dds:weight-by-alpha", "true")
            FrontImage.Settings.SetDefine("dds:cluster-fit", "true")
            FrontImage.Settings.AntiAlias = True
            FrontImage.ColorType = ColorType.TrueColor
            FrontImage.ColorSpace = ColorSpace.sRGB

            FrontImage.Write(SkyBox_Front)
            BackgroundWorker1.ReportProgress(16 / 6 * 100, 16)
            If BackgroundWorker1.CancellationPending Then
                e.Cancel = True

                Cancel_Create(SkyBox_Name & "_F.dds")
                Exit Sub
            End If
        End Using

After further tests i have found that the Compression method is DXT5 magick.net does not use BC7 ! 
BC7 is listed as a compression format but i cannot get Magick.net to use it, any suggestions?

**Comment by thegreatgramcracker** [comment]
I believe BC7 is only a format it can read and decompress, but it only compresses things in to BC1 or BC3, likely because other formats have very slow compression. Basically it will compress anything with alpha as BC3 and anything without as BC1. I recommend using [Direct X Tex Net](https://github.com/deng0/DirectXTexNet) to decompress your image in to a raw format (like R8G8B8A8) and then reading that with the compression define set to "none", then when you are done using ImageMagick use Direct X Tex to recompress it to it's original format.
